test/gtest/type-builder.cpp

```diff
@@ -457,3 +457,30 @@ TEST_F(IsorecursiveTest, CanonicalizeTypesBeforeSubtyping) {
   auto result = builder.build();
   EXPECT_TRUE(result);
 }
+
+static void testCanonicalizeBasicTypes() {
+  TypeBuilder builder(5);
+
+  Type externref = builder.getTempRefType(builder[0], Nullable);
+  Type externrefs = builder.getTempTupleType({externref, externref});
+
+  builder[0] = HeapType::ext;
+  builder[1] = Struct({Field(externref, Immutable)});
+  builder[2] = Struct({Field(Type::externref, Immutable)});
+  builder[3] = Signature(externrefs, Type::none);
+  builder[4] = Signature(Type({Type::externref, Type::externref}), Type::none);
```

Comment:
Which Type constructor is called here? Reading wasm-type.h I can't figure that out.

Reply:
`Type(std::initializer_list<Type>)`, which constructs a tuple type.

Reply:
I see, thanks. I've never really understood that part of C++ I guess...

Reply:
I removed the `Type` here to bring it more in line with how we use this in other places.